utils/layer_tree_manager.py

```diff
@@ -177,7 +180,15 @@ def _on_set_schematisation(self, something, filename=""):
         name = self.model_layergroup_basename + "/".join((split_dir[-1], split[-1]))
 
         # adjust spatialite for correct visualization of layers
-        threedi_db = ThreediDatabase({"db_path": filename})
+        db_settings = {"db_path": filename}
+        model_checker_db = threedi_database.ThreediDatabase(db_settings)
+        lib_version, file_version = get_spatialite_version(model_checker_db)
+        spatialite_version_upgrade_needed = True if file_version == 3 and lib_version in (4, 5) else False
```

Comment:
I think upgrade is only needed in case of spatialite 5 library:

```suggestion
        spatialite_version_upgrade_needed = file_version == 3 and lib_version == 5
```

Reply:
Discussed with Leendert, he thinks it is better to always show the warning. So I edited the suggestion